src/modules/DataLayer/services/datalayer.js

```diff
@@ -1,6 +1,19 @@
-import config from './mock-config';
+import Api from '@terralego/core/modules/Api';
 
-// Mock api to return the config
-export const fetchDatalayerConfig = () => config;
+let viewList = null;
 
-export default { fetchDatalayerConfig };
+export const fetchViewList = async () => {
+  if (viewList === null) {
+    const apiViews =  await Api.request('geolayer/view/');
+
+    // Object transformation from backend format
+    // to frontend one.
+    // Should be removed in a future release
+    viewList = Object.entries(apiViews)
+      .map(([key, { pk: id, ...rest }]) => ({ ...rest, id, key }))
+      .sort(({ id: idA }, { id: idB }) => idA - idB);
+  }
+
+  return viewList;
+};
+export default { fetchViewList };
```

Comment:
Why don't you use a provider and store `viewList` in a state ?
Then you can access to this viewList object everywhere with connecting the provider as HOC

Reply:
In a way setting up this HOC to fetch the list of available views is a temporary workaround to avoid storing _hard_ data in app.

In a near future, this `viewList` will be replaced by a react-admin resource (and a `<ReferenceInput />`). So there is no need to store in a global state that would not have any utility then.

Reply:
@dtrucs are you ok with that ?